pkg/cvo/metrics.go

```diff
@@ -162,3 +177,29 @@ func (m *operatorMetrics) Collect(ch chan<- prometheus.Metric) {
 		}
 	}
 }
+
+// mostRecentTimestamp finds the most recent change recorded to the status and
+// returns the seconds since the epoch.
+func mostRecentTimestamp(cv *configv1.ClusterVersion) int64 {
+	var latest time.Time
+	if len(cv.Status.History) > 0 {
+		latest = cv.Status.History[0].StartedTime.Time
+		if t := cv.Status.History[0].CompletionTime; t != nil {
+			if t.Time.After(latest) {
+				latest = t.Time
+			}
+		}
+	}
+	for _, condition := range cv.Status.Conditions {
+		if condition.LastTransitionTime.After(latest) {
+			latest = condition.LastTransitionTime.Time
+		}
+	}
+	if cv.CreationTimestamp.After(latest) {
```

Comment:
nit:
hmm, the placement for this check is odd, i think we are pretty sure that creation time is less than history and conditions...